src/adaptors/adaptor_common.h

```diff
@@ -0,0 +1,92 @@
+#ifndef __adaptor_common_h__
+#define __adaptor_common_h__ 1
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "entity.h"
+
+#include "qpid/dispatch/dispatch.h"
+#include "qpid/dispatch/threading.h"
+#include "qpid/dispatch/atomic.h"
+#include "qpid/dispatch/log.h"
+#include "qpid/dispatch/alloc_pool.h"
+
+typedef enum {
+    VERSION_HTTP1,
+    VERSION_HTTP2,
+} qd_http_version_t;
+
+typedef enum {
+    QD_AGGREGATION_NONE,
+    QD_AGGREGATION_JSON,
+    QD_AGGREGATION_MULTIPART
+} qd_http_aggregation_t;
+
+// Currently we only have TCP and HTTP (HTTP/1 and HTTP/2) adaptors
+typedef enum {
+    TCP,
+    HTTP
+} qd_adaptor_type_t;
+
+typedef struct qd_bridge_config_t qd_bridge_config_t;
+
+struct qd_bridge_config_t
+{
+    char              *name;
+    char              *host;
+    char              *port;
+    char              *address;
+    char              *site_id;
+    char              *host_port;
```

Comment:
host_port appears to the redundant with host and port.  Does the client really need to provide both forms of this data?

Reply:
host_port is set using host and port as seen here - https://github.com/skupperproject/skupper-router/blob/main/src/adaptors/tcp_adaptor.c#L1234